src/TurbViscSSTTAMSAlgorithm.C

```diff
@@ -0,0 +1,112 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2014 Sandia Corporation.                                    */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+// nalu
+#include <TurbViscSSTTAMSAlgorithm.h>
+#include <Algorithm.h>
+#include <FieldTypeDef.h>
+#include <Realm.h>
+
+// stk_mesh/base/fem
+#include <stk_mesh/base/BulkData.hpp>
+#include <stk_mesh/base/GetBuckets.hpp>
+#include <stk_mesh/base/MetaData.hpp>
+#include <stk_mesh/base/Part.hpp>
+#include <stk_mesh/base/Field.hpp>
+
+namespace sierra {
+namespace nalu {
+
+TurbViscSSTTAMSAlgorithm::TurbViscSSTTAMSAlgorithm(
+  Realm& realm, stk::mesh::Part* part)
+  : Algorithm(realm, part),
+    aOne_(realm.get_turb_model_constant(TM_aOne)),
+    betaStar_(realm.get_turb_model_constant(TM_betaStar)),
+    density_(NULL),
+    viscosity_(NULL),
+    tke_(NULL),
+    sdr_(NULL),
+    minDistance_(NULL),
+    dudx_(NULL),
+    tvisc_(NULL)
+{
+  // 2003 variant; basically, sijMag replaces vorticityMag
+  stk::mesh::MetaData& meta_data = realm_.meta_data();
+  density_ = meta_data.get_field<ScalarFieldType>(
+    stk::topology::NODE_RANK, "average_density");
```

Comment:
So the only difference with the TurbViscSST algorithm is that we use `average_density` and `average_dudx` here ?

Reply:
I consolidated the TurbViscSST implementation.